app/steps/ui/thankyou/index.js

```diff
@@ -8,9 +8,12 @@ class ThankYou extends Step {
         return '/thankyou';
     }
 
-    handleGet(ctx, formdata) {
-        ctx.softStop = this.anySoftStops(formdata, ctx);
-        return [ctx];
+    getContextData(req) {
+        const ctx = super.getContextData(req);
+        if (req.session.form.ccdCase && req.session.form.ccdCase.id) {
+            ctx.ccdReferenceNumber = req.session.form.ccdCase.id;
+        }
+        return ctx;
     }
```

Comment:
`getContextData()` gets called for both `handleGet()` and `handlePost()` so we'll need to remove `ctx.ccdReferenceNumber` in `action()` otherwise it'll get persisted when called by `handlePost()`.

Please also add unit tests for `action()`